src/does-match.js

```diff
@@ -37,45 +37,108 @@
       return string;
     }
 
-    var replaced = '';
-    for (var i = 0, size = string.length; i < size; i++) {
-      replaced += replaceDiacritics(string[i]);
-    }
+    var replaced = string.split('').map(replaceDiacritics);
 
-    return replaced;
+    return replaced.join('');
```

Comment:
I see your point, and I agree to some extend.

But my point is: this is a module that can be used both on client and server side, so most of the things you point out just don't apply. It should do one thing (and it does) and do it well (we're trying).

The most common use case for this lib, as I see it, is: filtering a collection as the user types, calling `doesMatch` for each item of the collection for each keystroke. If we have 100 items in the collection, with average text length of 100 characters, and we recalculate for every keystroke, there is some significative amount of processing that cannot be ignored. Specially with slow phones etc.

Besides, most of the examples that you said impact in page load. `doesMatch`, when used in client-side, is all about script execution time. So it should do it quickly, probably because there will be a lot of reflow that'll go along. And if it's used in server-side, using less CPU time is very desirable too.

The point is: I can measure and get to the conclusion that the performance impact is not significant, but I need to know before taking the decision. If we see, let's say, +50% of execution time, I think is not worth changing.